### ISSUE TITLE
Malicious Board when processing responses

### ISSUE THREAD

**Issue by gakonst** [celo-org/celo-threshold-bls-rs#56]
Looking at the following piece of code, it seems to me that the following attack is possible by the board:

1. Alice and Bob want to download responses
1. Board sends different responses to each, one where the status matrix is all `1s` and one where it's not
1. The set_statuses call will modify the status matrix of each participant
1. (wlog) Alice goes to Phase 3, while Bob sees no error responses and outputs his result

https://github.com/celo-org/celo-threshold-bls-rs/blob/c89dd0f190cf18b363abfffe5f65a54227a15651/crates/dkg-core/src/primitives/states.rs#L467

WDYT @nikkolasg @kobigurk

**Comment by kobigurk** [comment]
Yes, we definitely want a good broadcast channel. The coordinator approach should only be a stepping stone on the way and a trade-off if needed.
